ci/define-example-runner.sh

```diff
@@ -49,19 +49,23 @@ run_example() {
   local example=$2
   shift 2
   local arguments=$*
-  local program_name=$(basename ${program_path})
+  local program_name="$(basename "${program_path}")"
 
-  if [ ! -x ${program_path} ]; then
+  if [ ! -x "${program_path}" ]; then
     echo "${COLOR_YELLOW}[  SKIPPED ]${COLOR_RESET}" \
         " ${program_name} is not compiled"
     return
   fi
   log="$(mktemp -t "run_example.XXXXXX")"
   echo    "${COLOR_GREEN}[ RUN      ]${COLOR_RESET}" \
       " ${program_name} ${example} running"
-  echo ${program_path} ${example} ${arguments} >"${log}"
+  # We want ${arguments} to expand as separate parameters.
+  # shellcheck disable=SC2086
+  echo "${program_path}" "${example}" ${arguments} >"${log}"
   set +e
-  ${program_path} ${example} ${arguments} >>"${log}" 2>&1 </dev/null
+  # Ditto about ${arguments}.
+  # shellcheck disable=SC2086
+  ${program_path} "${example}" ${arguments} >>"${log}" 2>&1 </dev/null
```

Comment:
I too am unsure why `${program_path}` does not deserve quoting in such contexts.

Reply:
I was wrong. There's no reason not to quote it.